tests/secret-providers/aws_secretmanager_kiam/aws_secretmanager_kiam_test.go

```diff
@@ -0,0 +1,443 @@
+	postgreSQLUsername         = "test-user"
+	postgreSQLPassword         = "test-password"
+	postgreSQLDatabase         = "test_db"
+	postgreSQLConnectionString = fmt.Sprintf("postgresql://%s:%s@postgresql.%s.svc.cluster.local:5432/%s?sslmode=disable",
+		postgreSQLUsername, postgreSQLPassword, testNamespace, postgreSQLDatabase)
+	minReplicaCount = 0
+	maxReplicaCount = 2
+
+	awsRegion                = os.Getenv("TF_AWS_REGION")
+	awsAccessKeyID           = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey       = os.Getenv("TF_AWS_SECRET_KEY")
+	awsCredentialsSecretName = fmt.Sprintf("%s-credentials-secret", testName)
+	secretManagerSecretName  = fmt.Sprintf("connectionString-%d", GetRandomNumber())
+)
+
+type templateData struct {
+	TestNamespace                    string
+	DeploymentName                   string
+	ScaledObjectName                 string
+	TriggerAuthenticationName        string
+	SecretName                       string
+	PostgreSQLStatefulSetName        string
+	PostgreSQLConnectionStringBase64 string
+	PostgreSQLUsername               string
+	PostgreSQLPassword               string
+	PostgreSQLDatabase               string
+	MinReplicaCount                  int
+	MaxReplicaCount                  int
+	AwsRegion                        string
+	AwsCredentialsSecretName         string
+	SecretManagerSecretName          string
+	AwsAccessKeyID                   string
+	AwsSecretAccessKey               string
+}
+
+const (
+	deploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: postgresql-update-worker
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: postgresql-update-worker
+  template:
+    metadata:
+      labels:
+        app: postgresql-update-worker
+    spec:
+      containers:
+      - image: ghcr.io/kedacore/tests-postgresql
+        imagePullPolicy: Always
+        name: postgresql-processor-test
+        command:
+          - /app
+          - update
+        env:
+          - name: TASK_INSTANCES_COUNT
+            value: "6000"
+          - name: CONNECTION_STRING
+            valueFrom:
+              secretKeyRef:
+                name: {{.SecretName}}
+                key: postgresql_conn_str
+`
+
+	secretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+type: Opaque
+data:
+  postgresql_conn_str: {{.PostgreSQLConnectionStringBase64}}
+`
+	awsCredentialsSecretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.AwsCredentialsSecretName}}
+  namespace: {{.TestNamespace}}
+type: Opaque
+data:
+  AWS_ACCESS_KEY_ID: {{.AwsAccessKeyID}}
+  AWS_SECRET_ACCESS_KEY: {{.AwsSecretAccessKey}}
+`
+
+	triggerAuthenticationTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthenticationName}}
+  namespace: {{.TestNamespace}}
+spec:
+  awsSecretManager:
+ 	podIdentity:
+  	  provider: aws-kiam
```

Comment:
don't use aws-kiam (because we will deprecate it soon), use `aws-eks` instead please

Reply:
Alright.

Reply:
(and rename the stuff)